compiler/optimizer/OMRCFGSimplifier.cpp

```diff
@@ -169,6 +156,594 @@ bool TR_CFGSimplifier::simplifyBooleanStore()
+      return false;
+
+   _cfg->invalidateStructure();
+
+   TR::Node *condition = TR::Node::create(compareNode, compareNode->getOpCode().convertIfCmpToCmp(), 2,
+         compareNode->getFirstChild(),
+         compareNode->getSecondChild());
+
+   treeCursor = toCheck->getEntry()->getNextTreeTop();
+   while (treeCursor->getNode()->getOpCode().isStoreDirect()
+          && !treeCursor->getNode()->getOpCode().isWrtBar())
+      {
+      TR::Node *storeNode = treeCursor->getNode();
+      TR::Node *trueValue = triangle1 ? TR::Node::createWithSymRef(comp()->il.opCodeForDirectLoad(storeNode->getDataType()), 0, storeNode->getSymbolReference()) : (needToDuplicateTree ? storeNode->getFirstChild()->duplicateTree() : storeNode->getFirstChild());
+      TR::Node *falseValue = triangle1 ? (needToDuplicateTree ? storeNode->getFirstChild()->duplicateTree() : storeNode->getFirstChild()) : TR::Node::createWithSymRef(comp()->il.opCodeForDirectLoad(storeNode->getDataType()), 0, storeNode->getSymbolReference());
+
+      TR::Node *select = TR::Node::create(storeNode, comp()->il.opCodeForTernarySelect(storeNode->getDataType()), 3);
+      if (trace())
+         traceMsg(comp(), "Created ternary node n%dn\n", select->getGlobalIndex());
+
+      select->setAndIncChild(0, condition);
+      select->setAndIncChild(1, trueValue);
+      select->setAndIncChild(2, falseValue);
+      TR::TreeTop *insTree = TR::TreeTop::create(comp(), TR::Node::createWithSymRef(storeNode, storeNode->getOpCodeValue(), 1, select, storeNode->getSymbolReference()));
+
+      if (storeNode->getOpCodeValue() == TR::astore && storeNode->isHeapificationStore())
+         insTree->getNode()->setHeapificationStore(true);
+
+      compareTree->insertBefore(insTree);
+      treeCursor = treeCursor->getNextTreeTop();
+      }
+
+   if (triangle1) {
+      _cfg->removeEdge(_block, _next1);
+      if (_block->getNextBlock() != _next2)
+         {
+         TR::Node *gotoNode = TR::Node::create(compareNode, TR::Goto, 0);
+         gotoNode->setBranchDestination(_next2->getEntry());
+         _block->append(TR::TreeTop::create(comp(), gotoNode));
+         }
+   } else {
+      _cfg->removeEdge(_block, _next2);
+      if (_block->getNextBlock() != _next2)
+         {
+         TR::Node *gotoNode = TR::Node::create(compareNode, TR::Goto, 0);
+         gotoNode->setBranchDestination(_next1->getEntry());
+         _block->append(TR::TreeTop::create(comp(), gotoNode));
+         }
+   }
+   TR::TransformUtil::removeTree(comp(), compareTree);
+   if (trace())
+      traceMsg(comp(), "End simplifyCondStoreSequence.\n");   
+   TR::DebugCounter::incStaticDebugCounter(comp(), TR::DebugCounter::debugCounterName(comp(), "cfgSimpMovSeq/%d/(%s)", count, comp()->signature()));
+   return true;
+   }
+
+bool OMR::CFGSimplifier::simplifySimpleStore(bool needToDuplicateTree)
+   {
+   static char *disableSimplifySimpleStore = feGetEnv("TR_disableSimplifySimpleStore");
```

Comment:
Is there a reason for using an environment variable instead of an option here?

Reply:
Commented previously.